contracts/GnosisSafePersonalEdition.sol

```diff
@@ -0,0 +1,132 @@
+pragma solidity 0.4.23;
+import "./GnosisSafe.sol";
+import "./MasterCopy.sol";
+
+
+/// @title Gnosis Safe Personal Edition - A multisignature wallet with support for confirmations using signed messages based on ERC191.
+/// @author Stefan George - <stefan@gnosis.pm>
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract GnosisSafePersonalEdition is MasterCopy, GnosisSafe {
+
+    string public constant NAME = "Gnosis Safe Personal Edition";
+    string public constant VERSION = "0.0.1";
+    
+    uint256 internal constant BASE_TX_GAS_COSTS = 21000;
+    uint256 internal constant PAYMENT_GAS_COSTS = 10000;
+    // The maximum price for data payload is 68 (non-zero byte), but the avera is way lower, therefore we use 40
+    uint256 internal constant DATA_BYTE_GAS_COSTS = 40;
+
+    event ExecutionFailed();
+
+    uint256 public nonce;
+
+    /// @dev Allows to execute a Safe transaction confirmed by required number of owners.
+    /// @param to Destination address of Safe transaction.
+    /// @param value Ether value of Safe transaction.
+    /// @param data Data payload of Safe transaction.
+    /// @param operation Operation type of Safe transaction.
+    /// @param safeTxGas Gas that should be used for the Safe transaction.
+    /// @param gasPrice Gas price that should be used for the payment calculation.
+    /// @param v Array of signature V values sorted by owner addresses.
+    /// @param r Array of signature R values sorted by owner addresses.
+    /// @param s Array of signature S values sorted by owner addresses.
+    function payAndExecuteTransaction(
+        address to, 
+        uint256 value, 
+        bytes data, 
+        Enum.Operation operation, 
+        uint256 safeTxGas,
+        uint256 gasPrice,
+        uint8[] v, 
+        bytes32[] r, 
+        bytes32[] s
+    )
+        public
+    {
+        uint256 startGas = gasleft();
+        require(address(this).balance - value >= totalGasCosts(safeTxGas) * gasPrice);
+        checkHash(getTransactionHash(to, value, data, operation, safeTxGas, gasPrice, nonce), v, r, s);
+        // Increase nonce and execute transaction.
+        nonce++;
+        require(gasleft() - PAYMENT_GAS_COSTS >= safeTxGas);
```

Comment:
Is this check really necessary? The tx service should be able to verify that this require statement will always be satisfied at the time the transaction is sent.

Reply:
This is so that the service cannot set a too low total gas that makes the user transaction fail, but still allow a successful transfer of funds

Reply:
Oh right, because execute(to, value, data, operation, safeTxGas) wont revert if safeTxGas < gasleft() right?

Reply:
yes